doc/inband-telemetry/SAI-Proposal-Advanced-Telemetry-draft.md

```diff
@@ -0,0 +1,740 @@
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_REPORT_VIRTUAL_ROUTER_ID,
+
+    /**
+     * @brief Telemetry report truncate size
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @default 256
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_REPORT_TRUNCATE_SIZE,
+
+    /**
+     * @brief Telemetry report UDP destination port
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @default: 9999
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_REPORT_UDP_DST_PORT,
+
+    /**
+     * @brief Telemetry flow state clear cycle
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_FLOW_STATE_CLEAR_CYCLE,
+
+    /**
+     * @brief Latency sensitivity for flow state change detection
+     *
+     * @type sai_uint8_t
+     * @flags CREATE_AND_SET
+     * @default 15
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_LATENCY_SENSITIVITY,
+
+    /**
+     * @brief INT sink downstream ports
+     *
+     * @type sai_object_list_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_INT_SINK_PORT_LIST,
+
+    /**
+     * @brief Reserved DSCP value for INT over L4
+     *
+     * @type sai_ternary_field_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_SWITCH_ATTR_TELEMETRY_INT_L4_DSCP,
+
+} sai_switch_attr_t;
+~~~
+
+## Changes to `saiacl.h` for INT, Postcard and MoD watchlist
+~~~cpp
+typedef enum _sai_acl_action_type_t
+{
+    ......
+    /** Enable mirror on drop */
+    SAI_ACL_ACTION_TYPE_TELEMETRY_MOD_ENABLE,
+
+    /** Enable in-band network telemetry */
+    SAI_ACL_ACTION_TYPE_TELEMETRY_INT_ENABLE,
+
+    /** INT session ID */
+    SAI_ACL_ACTION_TYPE_TELEMETRY_INT_SESSION,
+
+    /** Enable packet postcard */
+    SAI_ACL_ACTION_TYPE_TELEMETRY_POSTCARD_ENABLE,
+
+    /** Telemetry flow sample percent within matched flow space */
+    SAI_ACL_ACTION_TYPE_TELEMETRY_FLOW_SAMPLE_PERCENT,
+
+    /** Report every packet for the matched flow */
+    SAI_ACL_ACTION_TYPE_TELEMETRY_REPORT_ALL,
+
+} sai_acl_action_type_t;
+
+typedef enum _sai_acl_table_attr_t
+{
+    ......
+    /**
+     * @brief Tunnel VNI
+     *
+     * @type bool
+     * @flags CREATE_ONLY
+     * @default false
+     */
+    SAI_ACL_TABLE_ATTR_FIELD_TUNNEL_VNI,
+
+    /**
+     * @brief Inner EtherType
+     *
+     * @type bool
+     * @flags CREATE_ONLY
+     * @default false
+     */
+    SAI_ACL_TABLE_ATTR_FIELD_INNER_ETHER_TYPE,
+
+    /**
+     * @brief Inner IP Protocol
+     *
+     * @type bool
+     * @flags CREATE_ONLY
+     * @default false
+     */
+    SAI_ACL_TABLE_ATTR_FIELD_INNER_IP_PROTOCOL,
+
+    /**
+     * @brief Inner L4 Src Port
+     *
+     * @type bool
+     * @flags CREATE_ONLY
+     * @default false
+     */
+    SAI_ACL_TABLE_ATTR_FIELD_INNER_L4_SRC_PORT,
+
+    /**
+     * @brief Inner L4 Dst Port
+     *
+     * @type bool
+     * @flags CREATE_ONLY
+     * @default false
+     */
+    SAI_ACL_TABLE_ATTR_FIELD_INNER_L4_DST_PORT,
```

Comment:
do we also need INNER_SRC_IP and INNER_DST_IP as well?

Reply:
These are already in the saiacl.h